applications/ShapeOptimizationApplication/custom_utilities/damping/damping_utilities.h

```diff
@@ -0,0 +1,396 @@
+    void InitalizeDampingFactorsToHaveNoInfluence()
+    {
+        for (ModelPart::NodeIterator node_i = mrModelPartToDamp.NodesBegin(); node_i != mrModelPartToDamp.NodesEnd(); ++node_i)
+        {
+            node_i->SetValue(DAMPING_FACTOR_X,1.0);    
+            node_i->SetValue(DAMPING_FACTOR_Y,1.0);  
+            node_i->SetValue(DAMPING_FACTOR_Z,1.0);  
+        } 
+    }
+
+    // --------------------------------------------------------------------------
+    void SetDampingFactorsForAllDampingRegions()
+    {
+        std::cout << "\n> Starting to prepare damping..." << std::endl;
+
+        // Loop over all regions for which damping is to be applied
+        for (unsigned int regionNumber = 0; regionNumber < len(mrDampingRegions); regionNumber++)
+        {
+            std::string dampingRegionSubModelPartName = mDampingSettings[regionNumber]["sub_model_part_name"].GetString();
+            ModelPart& dampingRegion = extractModelPart( mrDampingRegions[dampingRegionSubModelPartName] );
+
+            bool dampX = mDampingSettings[regionNumber]["damp_X"].GetBool();
+            bool dampY = mDampingSettings[regionNumber]["damp_Y"].GetBool();
+            bool dampZ = mDampingSettings[regionNumber]["damp_Z"].GetBool();
+            std::string dampingFunctionType = mDampingSettings[regionNumber]["damping_function_type"].GetString();
+            double dampingRadius = mDampingSettings[regionNumber]["damping_radius"].GetDouble();
+
+            DampingFunction::Pointer mpDampingFunction = CreateDampingFunction( dampingFunctionType, dampingRadius );
+
+            // Loop over all nodes in specified damping sub-model part 
+            for ( ModelPart::NodeIterator node_itr = dampingRegion.NodesBegin(); node_itr != dampingRegion.NodesEnd(); ++node_itr )
+            {
+                ModelPart::NodeType& currentDampingNode = *node_itr;
+                NodeVector neighbor_nodes( mMaxNeighborNodes );
+                DoubleVector resulting_squared_distances( mMaxNeighborNodes,0.0 );
+                unsigned int number_of_neighbors = mpSearchTree->SearchInRadius( currentDampingNode,
+                                                                                 dampingRadius, 
+                                                                                 neighbor_nodes.begin(),
+                                                                                 resulting_squared_distances.begin(), 
+                                                                                 mMaxNeighborNodes );
+
+                ThrowWarningIfNodeNeighborsExceedLimit( currentDampingNode, number_of_neighbors );                                                                               
+
+                // Loop over all nodes in radius (including node on damping region itself)
+                for(unsigned int j_itr = 0 ; j_itr<number_of_neighbors ; j_itr++)
+                {
```

Comment:
The reason to avoid unsigned `int` is that it tends to give OpenMP problems if the compiler only supports the standard up to 2.0  (aka: visual studio).

In this particular case its ok to leave it like that. Just have in  mind that if you intend to use OpenMP in this loop you will have to change it to `int`. Something like:

```C++
#pragma omp parallel for 
for(int j_itr = 0 ; j_itr < static_cast<int>(number_of_neighbors) ; j_itr++)
```